docs/flink-ddl.md

```diff
@@ -160,20 +161,36 @@ Table create commands support the commonly used [Flink create clauses](https://n
 * `COMMENT 'table document'` to set a table description.
 * `WITH ('key'='value', ...)` to set [table configuration](../configuration) which will be stored in Iceberg table properties.
 
-Currently, it does not support computed column, primary key and watermark definition etc.
+Currently, it does not support computed column and watermark definition etc.
+
+#### `PRIMARY KEY`
+
+Primary key constraint can be declared for a column or a set of columns, which must be unique and do not contain null.
+It's required for [`UPSERT` mode](../flink/flink-writes.md#upsert).
+
+```sql
+CREATE TABLE `hive_catalog`.`default`.`sample` (
+    id BIGINT COMMENT 'unique id',
+    data STRING NOT NULL,
+    PRIMARY KEY(`id`) NOT ENFORCED
+) 
+WITH ('format-version'='2');
+```
 
-### `PARTITIONED BY`
+#### `PARTITIONED BY`
 
 To create a partition table, use `PARTITIONED BY`:
 
 ```sql
 CREATE TABLE `hive_catalog`.`default`.`sample` (
     id BIGINT COMMENT 'unique id',
-    data STRING
-) PARTITIONED BY (data);
+    data STRING NOT NULL
+) 
+PARTITIONED BY (data) 
```

Comment:
I think the `PARTITIONED BY (data) WITH ('format-version'='2');` should be appended to the previous line. Same for the other places that were updated

Reply:
I've put `with` on the same line as `)` in other places but for `PARTITIONED BY` it's on a separate line if we follow [Flink's doc](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/create/)